exchange/exchange.go

```diff
@@ -660,7 +660,7 @@ func applyCategoryMapping(ctx context.Context, requestExt *openrtb_ext.ExtReques
 						// An older bid from a different seatBid we've already finished with
 						oldSeatBid := (seatBids)[dupe.bidderName]
 						if len(oldSeatBid.bids) == 1 {
-							seatBidsToRemove = append(seatBidsToRemove, bidderName)
+							seatBidsToRemove = append(seatBidsToRemove, dupe.bidderName)
```

Comment:
![image](https://user-images.githubusercontent.com/15863840/91202251-8740a180-e6cf-11ea-83ac-74072c935bf8.png)

Current test cases, including the recently added `TestCategoryMappingTwoBiddersOneBidEachNoCategorySamePrice`, do not seem to assert the recently modified functionality in line 663. Can we add coverage?

Reply:
Good catch.

Reply:
Execution of this part of cade is controlled by random number generation `rand.Intn(100)`:
```
if rand.Intn(100) < 50 {
      dupeBidPrice = -1
} else {
     currBidPrice = -1
}
```
which always gives same value on the first local run. 

In CI unit tests run with race condition: `./validate.sh --nofmt --cov --race 10` and without fix in exchange.go CI build fails with error from newly added unit test: 
https://travis-ci.org/github/prebid/prebid-server/jobs/721078338

To make it more explicitly I added for loop to execute `applyCategoryMapping` in test several times in a row to "force" `rand.Intn(100)` return different values.